tests/bytecode.rs

```diff
@@ -0,0 +1,27 @@
+#[cfg(test)]
+mod tests {
+    use core::panic;
+
+    use yaiwr::{instruction::Instruction, Calc};
+
+    #[test]
+    fn bytecode() {
+        let ast = Calc::from_str("1*2").unwrap();
+        let bytecode = &mut vec![];
+        let bytecode = Calc::to_bytecode(ast, bytecode);
+        assert_eq!(bytecode.len(), 3);
+
+        match bytecode[0] {
+            Instruction::Push { value } => assert_eq!(value, 1),
+            _ => panic!("Unexpected Instruction::Push value"),
+        }
+        match bytecode[1] {
+            Instruction::Push { value } => assert_eq!(value, 2),
+            _ => panic!("Unexpected Instruction::Push value"),
+        }
+        match bytecode[2] {
+            Instruction::Mul => {}
+            _ => panic!("expected Instruction::Mul instruction"),
+        }
+    }
```

Comment:
Should we add tests for overflows?

Reply:
Good idea. Note that Rust has different overflow behaviour in debug and release mode so CI will want do both `cargo test` *and* `cargo test --release` to make sure that both paths are tested.

Reply:
added overflow tests :point_right: https://github.com/softdevteam/pavel.yaiwr/pull/10/commits/f8fa3bc80308f5b9ce4057776ee43a10d2febbaf
Currently testing for `u64` since its hardcoded in `Calc` implementation, can be change to generics in the future

Reply:
He's using explicit calls to checked arithmetic at the moment, so there's only one mode to test.

Reply:
LGTM. Was gonna suggest you test underflow, but your language can't express a negative product, so :smile: